bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,86 @@
-import os
-import argparse
 import sys
-import json
-
 import clang.cindex as clang
 
+import utils
+
+
+def print_node(cursor, depth):
+    print("-" * depth, cursor.location.file, f"L{cursor.location.line} C{cursor.location.column}", cursor.kind.name, cursor.spelling)
+
 
-class bcolors:
-    HEADER = "\033[95m"
-    OKBLUE = "\033[94m"
-    OKGREEN = "\033[92m"
-    WARNING = "\033[93m"
-    FAIL = "\033[91m"
-    ENDC = "\033[0m"
-    BOLD = "\033[1m"
-    UNDERLINE = "\033[4m"
-
-
-def print_node(cursor, lines, depth):
-    file = cursor.location.file
-    line, column = cursor.location.line, cursor.location.column
-    result_type = (
-        cursor.result_type.spelling if cursor.result_type.spelling is not "" else "None"
-    )
-    print(
-        "-" * depth,
-        file,
-        f"L{line} C{column}",
-        bcolors.BOLD + cursor.kind.name + bcolors.ENDC,
-        bcolors.OKBLUE + cursor.spelling + bcolors.ENDC,
-        bcolors.OKGREEN + cursor.access_specifier.name + bcolors.ENDC,
-        bcolors.OKGREEN + "->" + bcolors.ENDC,
-        bcolors.OKGREEN + result_type + bcolors.ENDC,
-    )
-
-
-def node_in_this_file(node, file_name):
-    return node.location.file and node.location.file.name == file_name
-
-
-def walk_and_print(cursor, filter, lines, this_filename, depth):
+def is_node_in_this_file(node, filename):
+    return node.location.file and node.location.file.name == filename
+
+
+def walk_and_print(cursor, this_filename, depth):
```

Comment:
What are we walking?

Reply:
The AST :joy: 
Change to `traverse_and_print`?

Reply:
Changed to `print_ast`